src/core/tsi/ssl_transport_security.cc

```diff
@@ -619,10 +619,15 @@ static tsi_result x509_store_load_certs(X509_STORE* cert_store,
       sk_X509_NAME_push(*root_names, root_name);
       root_name = nullptr;
     }
+    ERR_clear_error();
     if (!X509_STORE_add_cert(cert_store, root)) {
-      gpr_log(GPR_ERROR, "Could not add root certificate to ssl context.");
-      result = TSI_INTERNAL_ERROR;
-      break;
+      size_t error = ERR_get_error();
```

Comment:
Nit: `size_t` isn't really the right type here. OpenSSL returns an `unsigned long`, so that's probably what you want. (BoringSSL uses `uint32_t` because it really should be a fixed-width integer and Windows makes `long` 32-bit. But `unsigned long` will work with both.

Reply:
Done